File: ROS/ros2_ws/src/my_cpp_pkg/src/HardwareStatusPublisher.cpp
#include "my_robot_interfaces/msg/hardware_status.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace my_robot_interfaces::msg;

class HardwareStatusPublisherNode : public rclcpp::Node
{
  public:

    HardwareStatusPublisherNode()
    : Node("hardware_status_publisher")
    {
      m_publisher = this->create_publisher<HardwareStatus>("hardware_status", 10);
      m_timer     = this->create_wall_timer(
        std::chrono::seconds(1),
        std::bind(&HardwareStatusPublisherNode::publishStatus, this));

      RCLCPP_INFO(this->get_logger(), "C++ hardware status publisher has started.");
    }

  private:

    void
    publishStatus()
    {
      HardwareStatus msg   = HardwareStatus();
      msg.temperature      = 40;
      msg.are_motors_ready = false;
      msg.debug_message    = "Not OK";

      m_publisher->publish(msg);
    }

  private:

    rclcpp::Publisher<HardwareStatus>::SharedPtr m_publisher;
    rclcpp::TimerBase::SharedPtr                 m_timer;
};

int
main(int argc, char** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<HardwareStatusPublisherNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/hardware_status_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus
 
 
class HardwareStatusPublisherNode(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")

        self.status_publisher_ = self.create_publisher(HardwareStatus, "hardware_status", 10)
        self.timer_ = self.create_timer(1, self.publish_hardware_status)

        self.get_logger().info("Hardware status publisher has started.")

    def publish_hardware_status(self):
        msg = HardwareStatus()
        msg.temperature = 50
        msg.are_motors_ready = True
        msg.debug_message = "Ok"

        self.status_publisher_.publish(msg)

 
def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()